### PR TITLE
Document how to use the Checker Framework with javac.

### PR DESCRIPTION
**Pull request by smillst**
I've added instructions for both Java 8 and Java 11.  ~~We'll want to comment out the Java 11 instructions until we are ready to release 3.0.0.~~ Next release will be 3.0.0, so I'm merging everything.

Fixes the last task of #2749.

### CODE REVIEW

File: docs/manual/external-tools.tex [hunk 1]
```diff
@@ -1004,6 +1005,99 @@
...
+-Xbootclasspath/p:$CHECKERFRAMEWORK/checker/dist/jdk8.jar \
+-cp $CHECKERFRAMEWORK/checker/dist/checker-qual.jar \
+-processorpath $CHECKERFRAMEWORK/checker/dist/checker.jar \
+-processor org.checkerframework.checker.nullness.NullnessChecker \
+-source 8 -target 8
+\end{Verbatim}
+
+Below is an explanation of each argument.
+\begin{enumerate}
+\item \<-J-Xbootclasspath/p:\$CHECKERFRAMEWORK/checker/dist/javac.jar>: the
```
Comment (smillst) [inline docs/manual/external-tools.tex]: Would it make sense to add that without this, you'll get 
`error: SourceChecker.typeProcessingStart: unexpected Throwable (NoSuchMethodError); message: com.sun.tools.javac.code.Type.stripMetadata()Lcom/sun/tools/javac/code/Type;`  or should we put that elsewhere?

File: docs/manual/external-tools.tex [hunk 2]
```diff
@@ -1004,6 +1005,99 @@
...
+
+\subsectionAndLabel{Java 11}{java11-install}
+
+This section shows what arguments to pass to javac to run the Checker Framework using JDK 11.  These
+instructions should work on JDK 9 or newer, but we only test with JDK 11.
+
+For compiling non-modularized code:
+\begin{Verbatim}
+javac \
+-J--add-opens=jdk.compiler/com.sun.tools.javac.comp=ALL-UNNAMED \
```
Comment (smillst) [inline docs/manual/external-tools.tex]: Without this, you'll get 
```
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.checkerframework.javacutil.Resolver 
```

File: docs/manual/external-tools.tex (hunk 1 above)
Comment (smillst) [reply docs/manual/external-tools.tex]: Also, `java.lang.NoSuchFieldError: ANNOTATION_PROCESSOR_MODULE_PATH` if you are compiling with source 9+.

File: docs/manual/external-tools.tex [hunk 3]
```diff
@@ -1004,6 +1005,99 @@
...
+
+\item \<-processorpath \$CHECKERFRAMEWORK/checker/dist/checker.jar>:
+checker.jar must be on the processor path.
+
+\item \<-processor org.checkerframework.checker.nullness.NullnessChecker>:
+choose which checker to run by passing its fully qualified name as a processor.
+(Note, using this options means that javac will not search for annotation
+processors, but rather run only those specified here.)
+
+\item \<-source 8 -target 8>: Because using the ErrorProne \<javac.jar> turns
```
Comment (smillst) [inline docs/manual/external-tools.tex]: Also, java.lang.NoSuchFieldError: ANNOTATION_PROCESSOR_MODULE_PATH if you are compiling with source 9+.

File: docs/manual/external-tools.tex [hunk 4]
```diff
@@ -420,6 +421,131 @@
...
+The \<-J> indicates that this is a JVM argument rather than a compiler
+argument.
+
+The following exception is thrown if this argument is missing:
+\begin{Verbatim}
+error: SourceChecker.typeProcessingStart: unexpected Throwable (NoSuchMethodError);
+message: com.sun.tools.javac.code.Type.stripMetadata()Lcom/sun/tools/javac/code/Type;
+\end{Verbatim}
+
+If you are compiling your own checker, the following exception is thrown if this argument is missing:
```
[the remark this answers is not in the document]
Comment (smillst) [reply docs/manual/external-tools.tex]: No, I tried that and did not get an error.

**Comment by maggu2810** [comment]
Why is this information not visible on https://checkerframework.org/manual/#javac-installation

Can you please update the manual?

**Comment by mernst** [comment]
The URL you pointed at says (at the top), "Version 2.11.1 (1 Oct 2019)".
That URL will be updated when the Checker Framework is next released, which is planned for Friday, November 1, 2019.

**Pull request merged**